app/src/main/java/corp/kairos/adamastor/collector/CollectorService.java

```diff
@@ -0,0 +1,400 @@
+            case Display.STATE_DOZE_SUSPEND : result = true; break;
+            case Display.STATE_UNKNOWN : result = false; break;
+            default : result =false; break;
+        }
+
+        return result;
+    }
+
+    private boolean isMusicPlaying(){
+        AudioManager am = (AudioManager)this.getSystemService(Context.AUDIO_SERVICE);
+        return am.isMusicActive();
+    }
+
+    private int getPhoneCallState(){
+        int result;
+        TelephonyManager tm = (TelephonyManager)getSystemService(Context.TELEPHONY_SERVICE);
+        switch(tm.getCallState()){
+            case TelephonyManager.CALL_STATE_IDLE : result = 0;
+                break;
+            case TelephonyManager.CALL_STATE_OFFHOOK: result = 1;
+                break;
+            case TelephonyManager.CALL_STATE_RINGING: result = 2;
+                break;
+            default : result = 0;
+        }
+
+        return result;
+    }
+
+    private void checkLocationPermissions(){
+        Criteria criteria = new Criteria();
+        criteria.setAccuracy(Criteria.ACCURACY_FINE);
+
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            checkLocation = false;
+        }else{
+            checkLocation = true;
+        }
+
+        lastLocation = locationManager.getLastKnownLocation(locationManager.getBestProvider(criteria,true));
+
+        if(lastLocation == null){
+            checkLocation = false;
+        }
+
+        locationManager.requestLocationUpdates(
+                locationManager.getBestProvider(criteria,true),
+                1000,
+                10,
+                locationListener);
+
+    }
+
+    private String getUsername() {
+        AccountManager manager = AccountManager.get(this);
+        Account[] accounts = manager.getAccountsByType("com.google");
+        List<String> possibleEmails = new LinkedList<String>();
+        for (Account account : accounts) {
+            possibleEmails.add(account.name);
+        }
+
+        if (!possibleEmails.isEmpty() && possibleEmails.get(0) != null) {
+            String email = possibleEmails.get(0);
+            String[] parts = email.split("@");
+            if (parts.length > 0 && parts[0] != null)
+                return parts[0];
+            else
+                return "null";
+        } else
+            return "null";
+    }
+
+    private int getRingtoneMode(){
+        int result;
+        AudioManager am = (AudioManager)this.getSystemService(Context.AUDIO_SERVICE);
+        switch(am.getRingerMode()){
+            case AudioManager.RINGER_MODE_NORMAL : result = 0; break;
+            case AudioManager.RINGER_MODE_SILENT : result = 1; break;
+            case AudioManager.RINGER_MODE_VIBRATE : result = 2; break;
+            default : result = 0;
+        }
+
+        return result;
+    }
+
+
+    private synchronized boolean getAndLockCycle(){
```

Comment:
I'm missing the point of the purpose of this method and the `inCycle` instance variable. Won't the method always return true? Why are you synchronizing this if only this thread is accessing the variable?

Reply:
You are right. This was an artifact from before we figured out how to stop Android from creating multiple instances of the same Service. I removed the variable and method.